File: 주다애/Binary Searcth/prob1.java
import java.util.*;

class Solution {
    public long solution(int n, int[] times) {
        long answer = 0;
        // 이분탐색을 위한 정렬
        Arrays.sort(times);
        long min = times[0];
        // n은 int이므로 long으로 형변환 해주어야 한다.
        long max = times[times.length - 1] * (long) n;
        while(min <= max) {
            // 중간값
            long mid = (min + max) / 2;
            // 통과한 사람 수의 합
            long sum = 0;
            
            for(int time : times) {
                sum += (mid / time);
            }
            
            if(sum >= n) {
                answer = mid;
                max = mid - 1;
            }
            else {
                min = mid + 1;
            }
        }
        return answer;
    }
}

File: 전예진/BinarySearch/prob1.py
def solution(n, times):
    left, right = 1, max(times) * n
    
    while left <= right:
        mid = (left + right) // 2
        count = 0
        
        # mid 시간 동안 처리 가능한 인원 수 계산
        for time in times:
            count += mid // time
        
        # 처리 가능한 인원 수가 n보다 작으면 시간을 늘려야 함
        if count < n:
            left = mid + 1
        
        # 처리 가능한 인원 수가 n보다 크거나 같으면 시간을 줄여도 됨
        else:
            right = mid - 1
    
    return left
